### PR TITLE
 Add an option to display all available remotes

### PR DESCRIPTION
**Pull request by WLAraujo**
A command to list all the available remotes was added to the remote command
https://github.com/kworkflow/kworkflow/issues/729

### CODE REVIEW

File: src/kw_remote.sh [hunk 1]
```diff
@@ -311,5 +311,5 @@ function remote_help()
     '  remote rename <old> <new> - Rename remote' \
     '  remote --set-default=<remonte-name> - Set default remote' \
     '  remote (--verbose | -v) - be verbose' \
-    '  remote (--list | -l) - List all the remotes' 
+    '  remote (--list | -l) - List all the remotes'
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:314]: This commit should be squashed in the previous one.

File: src/kw_remote.sh [hunk 2]
```diff
@@ -11,6 +11,16 @@ function remote_main()
     exit 0
   fi
 
+  if [[ "$1" =~ -l|--list ]]; then
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:14]: The correct way to handle parameters is inside the parser function `parse_config_options`. Here, you should check for `options_values` variable.

File: src/kw_remote.sh [hunk 3]
```diff
@@ -11,6 +11,16 @@ function remote_main()
...
+    if [[ -f "local_remote_config_file" ]]; then
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:15]: I think you forget the `$` in front of this variable.

File: src/kw_remote.sh [hunk 4]
```diff
@@ -11,6 +11,16 @@ function remote_main()
...
+      cat "$local_remote_config_file"
+      exit 0
+    else
+      echo "There are no remotes"
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:19]: If you are not concatenating any bash variable, just use single-quotes.

File: src/kw_remote.sh [hunk 5]
```diff
@@ -300,5 +310,6 @@ function remote_help()
     '  remote remove <name> - Remove remote' \
     '  remote rename <old> <new> - Rename remote' \
     '  remote --set-default=<remonte-name> - Set default remote' \
-    '  remote (--verbose | -v) - be verbose'
+    '  remote (--verbose | -v) - be verbose' \
+    '  remote (--list | -l) - List all the remotes'
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh]: I think this should also be the default behavior for `kw config`.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi,
I added some comments. Please, also add unit tests to this PR.

**Comment by rodrigosiqueira** [comment]
Please, rebase this change against the latest unstable branch and update this PR.

File: report_output [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Report:
```
Comment (rodrigosiqueira) [inline report_output:1]: Why this file is here?

File: src/kw_remote.sh [hunk 7]
```diff
@@ -210,6 +215,18 @@ function rename_remote()
   fi
 }
 
+function list_remotes()
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:218]: Please, add documentation on top of this function. Check `deploy.sh` for a reference.

File: src/kw_remote.sh [hunk 8]
```diff
@@ -210,6 +215,18 @@ function rename_remote()
...
+{
+  # check if the remote config file exists
+  if [ -f "$local_remote_config_file" ]; then
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:221]: Please, use `[[` instead of `[`.

File: src/kw_remote.sh [hunk 9]
```diff
@@ -210,6 +215,18 @@ function rename_remote()
...
+    cat $local_remote_config_file
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:222]: Just `cat` a file does not look good. Let's copy `git remote -v` output, something like:
```
$kw remote --list
  origin root@something *
  ubu-tm root@anotherip
  arch-tm root@anotherxpto
```

In other words, you need to process this data a little bit.

File: src/kw_remote.sh [hunk 10]
```diff
@@ -210,6 +215,18 @@ function rename_remote()
...
+    exit 0
+  else
+    echo "There are no remotes"
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:225]: Please, don't use `echo` refer to https://kworkflow.org/content/codingstyle.html#use-printf-instead-of-echo

File: src/kw_remote.sh [hunk 11]
```diff
@@ -210,6 +215,18 @@ function rename_remote()
   fi
 }
 
+function list_remotes()
+{
+  # check if the remote config file exists
+  if [ -f "$local_remote_config_file" ]; then
+    cat $local_remote_config_file
+    exit 0
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:223]: prefer return.

File: src/kw_remote.sh [hunk 12]
```diff
@@ -256,6 +273,10 @@ function parse_remote_options()
         options_values['RENAME']=1
         shift
         ;;
+      list)
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:276]: You need to add `list` to the `long_options` variable. You also need to update the `bash_autocomplete.sh`.

File: src/kw_remote.sh [hunk 13]
```diff
@@ -299,6 +320,7 @@ function remote_help()
     '  remote add <name> <USER@IP:PORT> [--set-default] - Add new remote' \
     '  remote remove <name> - Remove remote' \
     '  remote rename <old> <new> - Rename remote' \
+    '  remote list - List all the remotes' \
```
Comment (rodrigosiqueira) [inline src/kw_remote.sh:323]: We can use `--list` here.

File: tests/kw_remote_test.sh [hunk 14]
```diff
@@ -423,6 +423,10 @@ function test_parse_remote_options()
   parse_remote_options rename origin xpto
   assert_equals_helper 'Request rename' "($LINENO)" "${options_values['RENAME']}" 1
   assert_equals_helper 'Remote options' "($LINENO)" "${options_values['PARAMETERS']}" 'origin xpto '
+
```
Comment (rodrigosiqueira) [inline tests/kw_remote_test.sh:426]: Please, add tests that cover all the new code.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
See my inline comments, and also update the commit message (use `git log src/kw_remote.sh` if you need some references).

I recommend you to read this page:
https://cbea.ms/git-commit/

**Comment by rodrigosiqueira** [comment]
@WLAraujo any update here?

**Comment by rodrigosiqueira** [comment]
Hi @WLAraujo , please address my previous comments and take a look at some of the links that I shared with you.

**Comment by WLAraujo** [comment]
Hi @rodrigosiqueira!
Sorry it took me so long to get back to you.
I'll try to take a look at the comments and make the changes today.

**Comment by rodrigosiqueira** [comment]
@WLAraujo could you address my previous comments and fix the shellcheck failure?

**Comment by rodrigosiqueira** [comment]
I closed this PR due to the lack of reply. If you have a better version of this change, feel free to re-open this PR.

**Pull request closed without merging**